modules/custom/express_permissions/express_permissions.module

```diff
@@ -811,42 +811,77 @@ function express_permissions_form_user_profile_form_alter(&$form, &$form_state)
+    if (count($current_site_owners) >= $site_owners_limit) {
+      form_set_error('roles', 'You have exceeded the allowed limit of '. ($site_owners_limit - 1) .' Site Owners. To grant the role to this user you must first remove it from someone else. Current Site Owners include: '.
+        implode(', ', _expr_get_realnames_without_test_users($current_site_owners)));
     }
   }
 
-  if ((in_array($config_managers->rid, $form_state['values']['roles'], TRUE) || $form_state['values']['addon_roles'][$config_managers->rid] === $config_managers->rid) && !isset($current_roles[$config_managers->rid])) {
-    $config_managers_role = express_permissions_get_site_owners('configuration_manager');
-    if ($config_managers_role['count'] >= $addon_limit) {
-      form_set_error("", "You have exceeded the allowed limit for Configuration Managers. To grant the role to this user you must first remove it from someone else. Current Configuration Managers include: " . $config_managers_role['list']);
+  if (!isset($current_roles[$config_manager_role->rid]) && in_array($config_manager_role->rid, $form_state['values']['roles'], TRUE)) {
+    $current_config_managers = express_permissions_get_users_in_role('configuration_manager');
+    if (count($current_config_managers) >= $addon_limit) {
+      form_set_error('roles', 'You have exceeded the allowed limit of '. ($addon_limit - 1) .' Configuration Managers. To grant the role to this user you must first remove it from someone else. Current Configuration Managers include: '.
+        implode(', ', _expr_get_realnames_without_test_users($current_config_managers)));
     }
   }
 
-  if ((in_array($access_managers->rid, $form_state['values']['roles'], TRUE) || $form_state['values']['addon_roles'][$access_managers->rid] === $access_managers->rid) && !isset($current_roles[$access_managers->rid])) {
-    $access_managers_role = express_permissions_get_site_owners('access_manager');
-    if ($access_managers_role['count'] >= $addon_limit) {
-      form_set_error("", "You have exceeded the allowed limit for Access Managers. To grant the role to this user you must first remove it from someone else. Current Access Managers include: " . $access_managers_role['list']);
+  if (!isset($current_roles[$access_manager_role->rid]) && in_array($access_manager_role->rid, $form_state['values']['roles'], TRUE)) {
+    $current_access_managers = express_permissions_get_users_in_role('access_manager');
+    if (count($current_access_managers) >= $addon_limit) {
+      form_set_error('roles', 'You have exceeded the allowed limit of '. ($addon_limit - 1) .' Access Managers. To grant the role to this user you must first remove it from someone else. Current Access Managers include: '.
+        implode(', ', _expr_get_realnames_without_test_users($current_access_managers)));
     }
   }
 }
 
+
+/**
+ * @param $users
+ *
+ * @return array
+ */
+function _expr_get_realnames_without_test_users($users) {
```

Comment:
There should be some suite of functions used for dealing with the test accounts. `cu_users_get_users('test_accounts')` might work.

Reply:
@jeor0980 is working on that as part of https://github.com/cuboulder/express/issues/2196